src/pint/observatory/observatories.py

```diff
@@ -1,970 +1,87 @@
-)
-TopoObs(
-    name="grao",
-    aliases=["grao"],
-    itrf_xyz=[6346273.531, -33779.7127, 634844.9454],
-    origin="""Imported from TEMPO2 observatories.dat 2021 June 7.""",
-)
-TopoObs(
-    name="srt",
-    aliases=["srt"],
-    tempo_code="z",
-    itoa_code="SR",
-    itrf_xyz=[4865182.766, 791922.689, 4035137.174],
-    origin="""Imported from TEMPO2 observatories.dat 2021 June 7.""",
-)
-
-# From Tempo 2021 June 8
-TopoObs(
-    name="quabbin",
-    tempo_code="2",
-    itoa_code="QU",
-    itrf_xyz=[1430913.3496148302, -4495711.383965823, 4278113.974517222],
-    origin="""Imported from TEMPO obsys.dat 2021 June 8.""",
-)
-TopoObs(
-    name="vla_site",
-    tempo_code="c",
-    itoa_code="V2",
-    itrf_xyz=[-1601135.5133304405, -5042005.480977412, 3554875.076856462],
-    origin="""Imported from TEMPO obsys.dat 2021 June 8.""",
-)
-TopoObs(
-    name="gb_20m_xyz",
-    itoa_code="G2",
-    itrf_xyz=[883772.7974, -4924385.5975, 3944042.4991],
-    origin="""Imported from TEMPO obsys.dat 2021 June 8.""",
-)
-TopoObs(
-    name="northern_cross",
-    tempo_code="d",
-    itoa_code="BO",
-    itrf_xyz=[4461242.882451464, 919559.8351226494, 4449633.220012489],
-    origin="""Imported from TEMPO obsys.dat 2021 June 8.""",
-)
+import pint.config
+from pint.observatory.topo_obs import TopoObs
+from pint.observatory import bipm_default
+
+observatories_json = pint.config.runtimefile("observatories.json")
+
+pint_env_var = "PINT_OBS_OVERRIDE"
+
+__all__ = ["observatories_json", "read_observatories"]
+
+
+def get_default_value(entry, key, default=None):
+    """Return default value if key is not in entry.  Otherwise return entry[key]
+
+    Parameters
+    ----------
+    entry : dict
+    key : str
+    default :
+
+    Returns
+    -------
+        value for assignment
+    """
+
+    return default if not key in entry else entry[key]
+
+
+# Default values for instantiating a new TopoObs object
+# They will be overridden by any values set in the JSON file
+default_keywords = {
+    "tempo_code": None,
+    "itoa_code": None,
+    "aliases": None,
+    "itrf_xyz": None,
+    "clock_file": "",
+    "clock_dir": "PINT",
+    "clock_fmt": "tempo",
+    "include_gps": True,
+    "include_bipm": True,
+    "bipm_version": bipm_default,
+    "origin": None,
+    "bogus_last_correction": False,
+}
+
+
+def read_observatories(filename=observatories_json, overwrite=False):
+    """Read observatory definitions from JSON and create TopoObs objects, registering them
+
+    Set `overwrite` to ``True`` if you want to re-read a file with updated definitions
+
+    Parameters
+    ----------
+    filename : str, optional
+    overwrite : bool, optional
+        Whether a new instance of an existing observatory should overwrite the existing one.
+
+
+    """
+    # read in the JSON file
+    observatories = json.load(open(filename))
+    for obsname in observatories:
+        keywords = {}
+        for keyword in default_keywords:
```

Comment:
Can you skip this loop and just make `keywords` be (a copy of?) `observatories[obsname]`? Then the normal python call mechanism will handle unspecified entries by giving them the default arguments specified in the TopoObs constructor.